tobiasHe/assignments/cyclicSort/cyclicSortEasy/cyclicSortEasy.js

```diff
@@ -4,14 +4,34 @@
 
 // Write a function to sort the objects in-place on their creation sequence number in O(n) and without any extra space. For simplicity, let’s assume we are passed an integer array containing only the sequence numbers, though each number is actually an object.
 
-const cyclic_sort = function (nums) {
-  // TODO: Write your code here
-  return nums;
+const cyclic_sort = function(nums) {
+	// TODO: Write your code here
+	// cyclic sort, sorts in place with swaps
+	// given object representation array from 1-n integers, sort it by using cyclic sort
+	// nums[0] = 3   nums[3-1]
+	// 3 1 5 4 2
+	// nums[0] = 3
+	// swap nums[0] to position 2 which is where 5 is
+	// 3 - 1 = 2
+	// nums[0] - 1 = 2
+	// j = 2  <--- used for position to swap nums[j] wapped with nums[0]
+	// if [nums[i] !== nums[j]]  [3 !== 5]
+	//    destructing swap [a,b] = [b,a]
```

Comment:
It is great that you walked through a problem with what you want your solution to do, but the approach would be clearer if it were written out in full sentences. Also edge cases are missing.